src/plugins/external_tool_support/src/blast_plus/align_worker_subtasks/BlastReadsSubTask.cpp

```diff
@@ -56,21 +56,40 @@ BlastReadsSubTask::BlastReadsSubTask(const QString &dbPath,
       readsNames(readsNames),
       reference(reference),
       minIdentityPercent(minIdentityPercent),
+      readIndex(0),
       storage(storage) {
     setMaxParallelSubtasks(AppContext::getAppSettings()->getAppResourcePool()->getIdealThreadCount());
+    tpm = Progress_Manual;
 }
 
 void BlastReadsSubTask::prepare() {
     QString tempPath = AppContext::getAppSettings()->getUserAppsSettings()->getCurrentProcessTemporaryDirPath();
     CHECK_EXT(!GUrlUtils::containSpaces(tempPath), setError(tr("The task uses a temporary folder to process the data. The folder path is required not to have spaces. "
                                                                "Please set up an appropriate path for the \"Temporary files\" parameter on the \"Directories\" tab of the UGENE Application Settings.")), );
 
-    foreach (const SharedDbiDataHandler &read, reads) {
-        BlastAndSwReadTask *subTask = new BlastAndSwReadTask(dbPath, read, reference, minIdentityPercent, readsNames[read], storage);
-        addSubTask(subTask);
+    // Avoid adding too many subtasks at once. Large numbers of subtasks may work task scheduler slow and freeze the UI.
+    int firstBucketSize = qMin(getMaxParallelSubtasks(), reads.size());
+    for (readIndex = 0; readIndex < firstBucketSize; readIndex++) {
+        const SharedDbiDataHandler &read = reads[readIndex];
+        addSubTask(new BlastAndSwReadTask(dbPath, read, reference, minIdentityPercent, readsNames[read], storage));
+    }
+}
+
+QList<Task *> BlastReadsSubTask::onSubTaskFinished(Task *task) {
+    QList<Task *> newSubtasks;
+    CHECK(!isCanceled(), newSubtasks);
```

Comment:
Check hasError as well. This method doesn't contain any code that should be executed in case of errors.

Reply:
Done